### DISCUSSION TITLE
Add Execution Time Measurement in Milliseconds for the data retrieval

### DISCUSSION THREAD

**Discussion by arkdev1** [Power-Components/livewire-powergrid#1389] [Ideas]
Hello,

I am creating this discussion as an enhancement for livewire-powergrid.
I'm suggest adding functionality to measure the execution time of the "data retrieval" in milliseconds. This feature would enable developers to accurately benchmark and optimize their applications, ensuring that we can maintain high performance and improve user experience.

Example:
![image](https://github.com/Power-Components/livewire-powergrid/assets/138211005/56486f98-1e46-4f4f-8f04-91f9fdd62e6b)

Measuring execution time in milliseconds provides the precision needed for fine-tuning performance, identifying bottlenecks, and making data-driven decisions about caching, database queries optimization, and more.

> **Implementation Consideration:**
> The feature could be implemented as a middleware or a service provider that hooks into Laravel's lifecycle, measuring the start and end times around the package's core operations. Utilizing Laravel's built-in functions for timing and memory usage, this could be a lightweight addition that offers profound insights.

**Comment by dansysanalyst** [comment]
There are some external tools to help developers to fine tune their applications, for example: LaraDumps (www.laradumps.dev).

In a production environment, I think this fits more the job of a Database/Server monitor.

Talking about the end user (if the feature is activated), I fail to see how reading such metrics could benefit them. Users could even complain that "yesterday it took 50ms and today 55ms", generating some headache. 🙃 

So, these are just opinions. You are welcome to submit a Pull Request, adding/improving any features you consider valuable.

All contributions are appreciated and fully credited!

**Reply by arkdev1** [reply]
Hi @dansysanalyst,
I agree, the end user, or a consumer user, will never need this information.
However, I find myself about to build an application for more technical users who work with large databases related to the world of wine cellars. They provide me with tables and views that are populated by their machine learning models, etc.

I would be more than happy to try creating a class that overrides the standard class; I would just ask you which class I could look at to start understanding how the grid retrieves data?

**Comment by luanfreitasdev** [comment]
I agree with @dansysanalyst, however, some things can be measured within powergrid (queries, time it took to retrieve the data). I had already thought about something like this but didn't want to put it in a view or generate more dependencies, so I decided to work with events. 

https://github.com/Power-Components/livewire-powergrid/pull/1393

**Reply by arkdev1** [reply]
Magnificent, I'll take a look at the pull request!
